src/expr/src/scalar/like_pattern.rs

```diff
@@ -105,7 +105,7 @@ use matcher::MatcherImpl;
 // This lint interacts poorly with `derivative` here; we are confident it generates
 // compatible `PartialOrd` and `Ord` impls. Unfortunately it also requires we introduce
 // this module to allow it.
-#[allow(clippy::incorrect_partial_ord_impl_on_ord_type)]
+#[allow(clippy::non_canonical_partial_ord_impl)]
```

Comment:
looks like locally you must have a version of Rust > 1.71. We shouldn't include this change as part of this PR, instead it'll get updated by https://github.com/MaterializeInc/materialize/pull/23320/files

Reply:
Fixed in f0a4a30d51, thanks!